base/irrationals.jl

```diff
@@ -179,6 +179,28 @@ for op in Symbol[:+, :-, :*, :/, :^]
     @eval $op(x::AbstractIrrational, y::AbstractIrrational) = $op(Float64(x),Float64(y))
 end
 *(x::Bool, y::AbstractIrrational) = ifelse(x, Float64(y), 0.0)
+TwicePrecision(x::AbstractIrrational) = TwicePrecision{Float64}(x)
+
+for op in (:+, :-)
```

Comment:
Why do we need new methods here?  What's wrong with the current `irrational + number` that calls `promote` to use the same precision as the output (rather than twice the precision)?

It seems like this will potentially slow down lots of existing code that uses irrational constants, just to get an extra ulp?

Reply:
If you need extra precision specifically for range constructors, you can define an extra-precision method that is called in that place only.

Reply:
> What's wrong with the current irrational + number that calls promote to use the same precision as the output (rather than twice the precision)?

I just feel wrong that
```julia
julia> π-Float64(π)
0.0

julia> π==Float64(π)
false
```

> It seems like this will potentially slow down lots of existing code that uses irrational constants.

It will. By design, shouldn't irrationals be as precise as needed? If one doesn't need extra precision, they can always force convert.

> If you need extra precision

I'm just trying to stay consistent with what already exist (Julia always generates `TwicePrecision` steps when using colons and floats), as the rules are not explicitly stated in the documentation. `TwicePrecision(π) = TwicePrecision(Float64(π),0.0)` also feels wrong, and implementing accurate `π-Float64(π)` fixes the problem.